internal/pkg/build/args/reader.go

```diff
@@ -35,12 +38,27 @@ func NewReader(src io.Reader, buildArgsMap map[string]string, defaultArgsMap map
 		return nil, err
 	}
 
+	// do templating
 	matches := buildArgsRegexp.FindAllSubmatchIndex(srcBytes, -1)
 	mapOfConsumedArgs := make(map[string]bool)
 	var buf bytes.Buffer
 	bufWriter := io.Writer(&buf)
 	i := 0
 	for _, m := range matches {
+		// find the last newline
+		newlineIdx := i
+		for start := m[0]; start >= newlineIdx; start-- {
+			if srcBytes[start] == '\n' {
+				newlineIdx = start
+				break
+			}
+		}
+
+		// check whether current line containing {{ VAR }} is commented line
+		if commentLineRegexp.Match(srcBytes[newlineIdx:m[0]]) {
```

Comment:
What if there's a comment toward the end of the line after there's a variable to expand?  Won't this skip the whole line?

Reply:
> What if there's a comment toward the end of the line after there's a variable to expand? Won't this skip the whole line?

do you mean this type of comment?

let's assume `OS_VER=1`
```
apt install {{ OS_VER }}#comment {{ OS_VER }}
```

it will be parsed as
```
apt install 1#comment {{ OS_VER }}
```